htdocs/takepos/takepos.php

```diff
@@ -58,7 +58,7 @@
 $categorie = new Categorie($db);
 $categories = $categorie->get_full_arbo('product');
 ?>
-var categories = JSON.parse( '<?php echo json_encode($categories);?>' );
+var categories = <?php echo json_encode($categories); ?>;
```

Comment:
json_encode return a string so if you remove JSON.parse, the var categories will be a string.
And later it is used as an array with categories[]
The goal of JSON.parse is to convert a json string into the array. Are you sure yo can remove this ?

Reply:
@eldy @andreubisquerra 

`$categories = $categorie->get_full_arbo('product')` return an array !
`json_encode($categories)` return already an array in json format : 

`[ "one", "two", "three" ]`

if you add quotation marks this is a string format : 

`var categories = '<?php echo json_encode($categories); ?>';`

and without quote this is an array : 

`var categories = <?php echo json_encode($categories); ?>;`

with the current code, there are errors if the category labels contain a quotation mark because it conflicts with the quotation marks around the json.

`var categories = JSON.parse( '<?php echo json_encode($categories);?>' );`

Reply:
Yeah that's right. The current code fails if there are quotes in the categories, but because in Spain we do not use quotation, no problem for us.
Good job!